overlord/snapshotstate/backend/backend_test.go

```diff
@@ -509,3 +509,53 @@ func (s *snapshotSuite) TestHappyRoundtrip(c *check.C) {
 		c.Check(ioutil.WriteFile(filepath.Join(info.DataDir(), "marker"), []byte("scribble\n"), 0644), check.IsNil, comm)
 	}
 }
+
+func (s *snapshotSuite) TestRestoreRoundtripDifferentRevision(c *check.C) {
+	if os.Geteuid() == 0 {
+		c.Skip("this test cannot run as root (runuser will fail)")
+	}
+	logger.SimpleSetup()
+
+	epoch := snap.E("42*")
+	info := &snap.Info{SideInfo: snap.SideInfo{RealName: "hello-snap", Revision: snap.R(42), SnapID: "hello-id"}, Version: "v1.33", Epoch: *epoch}
+	shID := uint64(12)
+
+	shw, err := backend.Save(context.TODO(), shID, info, nil, []string{"snapuser"})
+	c.Assert(err, check.IsNil)
+	c.Check(shw.Revision, check.Equals, info.Revision)
+
+	shr, err := backend.Open(backend.Filename(shw))
+	c.Assert(err, check.IsNil)
+	defer shr.Close()
+
+	c.Check(shr.Revision, check.Equals, info.Revision)
+	c.Check(shr.Name(), check.Equals, filepath.Join(dirs.SnapshotsDir, "12_hello-snap_v1.33_42.zip"))
+
+	// move the expected data to its expected place
+	for _, dir := range []string{
+		filepath.Join(s.root, "home", "snapuser", "snap", "hello-snap"),
+		filepath.Join(dirs.SnapDataDir, "hello-snap"),
+	} {
+		c.Check(os.Rename(filepath.Join(dir, "42"), filepath.Join(dir, "17")), check.IsNil)
+	}
+
+	newroot := c.MkDir()
+	c.Assert(os.MkdirAll(filepath.Join(newroot, "home", "snapuser"), 0755), check.IsNil)
+	dirs.SetRootDir(newroot)
+
+	var diff = func() *exec.Cmd {
```

Comment:
(nitpick^2) you could fold the `diff().Run()` directly into diff. I.e.: `return cmd.Run()`.

Reply:
I think I'll leave it for now: this is in the other PR